### DISCUSSION TITLE
How to install jupyterlab-github extension in jupyterlite?

### DISCUSSION THREAD

**Discussion by dbworker** [jupyterlite/jupyterlite#812] [Q&A] [answered]
Hi gurus,
Jupyterlite is really an amazing software, I like it.  
When I saw that [the official jupyterlite demo](https://jupyterlite.rtfd.io/en/latest/try/lab) has jupyterlite-github extension enabled, ![julite-github.png](https://cdn.jsdelivr.net/gh/dbworker/mypic/github-ext.png)  
I want to make it on my local jupyterlite too. After install github extension on my laptop, it was shown on jupyterlab sidebar, but not on jupyterlite.   
May i ask how to enable it on jupyterlite? Thanks.

**Comment by dbworker** [comment]
I had read "Add additional extensions to a JupyterLite website" in jupyterlite.readthedocs.io, and i can't understand why some extensions cound take effect on jupyterlite automatically without any installation steps, such as those extensions under jupyterlab/examples, so sometimes I doubt maybe there is easy way to install extension.

**Comment by bollwyvl** [comment, marked as answer]
After installing an extension in the outer environment, you'd need to rerun `jupyter lab build` to pick up any new extensions from `{sys.prefix/share/jupyter/labextensions`.

In this repos example, we do it "the hard way" by explicitly listing them to avoid some we don't want.

**Comment by dbworker** [comment]
@bollwyvl  Thanks :)
I test all the ways, and found jupyterlite don't copy `{sys.prefix}/share/jupyter/labextensions/@jupyterlab/github` to its `_output/extensions`,  except `@jupyterlab-examples/commands` extension.
```
~/dev/gdoc$ jupyter lab build
[LabBuildApp] JupyterLab 3.4.2
[LabBuildApp] Building in /usr/local/share/jupyter/lab
[LabBuildApp] Building jupyterlab assets (production, minimized)

➜  ~/dev/gdoc$ jupyter lite build
static:jupyter-lite.json
.  pre_status:static:jupyter-lite.json
    tarball:         jupyterlite-app-0.1.0-beta.12.tgz 8MB
    output:          /Users/bruce/dev/gdoc/_output
    lite dir:        /Users/bruce/dev/gdoc
    apps:            
    sourcemaps:      True
    unused packages: True
archive:archive
contents:contents
lite:jupyter-lite.json
mathjax:status
mimetypes:jupyter-lite.json
pyodide:pyodide
serve:contents
settings:overrides
translation:translation
.  status:archive:archive
[LiteBuildApp] No archive (yet): gdoc-jupyterlite.tgz
.  status:contents:contents
    contents: 0 files
.  status:lite:jupyter-lite.json
[LiteBuildApp]     jupyter-lite.(json|ipynb): 0 files
.  status:mathjax:status
     jupyter-server-mathjax: None 
     configured MathJax dir: /Users/bruce/dev/gdoc/traitlets.Undefined
     effective MathJax dir: None
.  status:mimetypes:jupyter-lite.json
    filetypes:         26 
.  status:pyodide:pyodide
     URL: None
 archive: []
   cache: 0 files
   local: 0 files
.  status:serve:contents
    url: http://127.0.0.1:8000/
    server: tornado
    headers:
.  status:settings:overrides
    overrides.json: 0
.  status:translation:translation
    translation files: 0 files
static:output_dir
-- pre_init:static:output_dir
static:unpack
.  init:static:unpack
federated_extensions:copy:ext:@jupyterlab-examples/commands
.  pre_build:federated_extensions:copy:ext:@jupyterlab-examples/commands
translation:copy
.  build:translation:copy
federated_extensions:patch
mimetypes:patch
.  post_build:federated_extensions:patch
.  post_build:mimetypes:patch
➜  ~/dev/gdoc$ 
```
My installed python packages are
```
jupyterlab                   3.4.2
jupyterlab-examples-commands 0.1.0       /Users/bruce/dev/jexample/commands
jupyterlab-github            3.0.1
jupyterlab-gitlab            3.0.0
jupyterlab-pygments          0.2.2
jupyterlab-server            2.15.0
jupyterlab-widgets           1.1.0
jupyterlite                  0.1.0b12
```

**Comment by bollwyvl** [comment]
Try deleting the _output and rerunning "lite build"

You should not need to be running "lab build"

**Comment by dbworker** [comment]
I delete and rebuild, and result is the same.  
Troubleshooting is not easy, i'll test on a brand new env.

**Comment by dbworker** [comment]
@bollwyvl  Finally get what you said "In this repos example, we do it `the hard way` by explicitly listing them", yes, {[the hard way](https://jupyterlite.readthedocs.io/en/latest/howto/configure/advanced/hard.html#)} doc tell user  
  manually copying  `{sys.prefix}/share/jupyter/labextensions/@jupyterlab/github`  
  to 
  `$YOUR_JUPYTERLITE/_output/extensions`,
and jupyterlite will add  extension info in file `jupyter-lite.json#federated_extensions` after restart, then github-extension take effect now.
Thanks
